src/Microsoft.Crank.PullRequestBot/Program.cs

```diff
@@ -142,13 +145,21 @@ private static async Task<int> Controller(BotOptions options)
                 return 1;
             }
 
+            CommentCutoffDate =  DateTime.Now.AddMinutes(0 - _options.Age);
+
             if (_options.GitHubBaseUrl != null)
             {
                 // GitHub Enterprise may require the client to be authenticated,
                 // rather than just use a different base address for the API.
                 await UpgradeAuthenticatedClient();
             }
 
+            // If authentication information is provided pre-authenticate since it will prevent some rate limiting exception for read-only API calls.
+            if (!String.IsNullOrEmpty(_options.AccessToken) || !String.IsNullOrEmpty(_options.AppId) || !String.IsNullOrEmpty(_options.AppId))
```

Comment:
Should this be an `else if` since the previous `else` might have authenticated already?

nit: Use lowercase `string`

Reply:
UpgradeAuthentication can be called multiple times, it's done later on in the code if we need to create new comments for instances, independently of these blocks.

Reply:
Sure, but why call it right after it was just called? Will that actually change anything?